src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -117,6 +119,9 @@ protected boolean selfUpdate() {
                 final String jarDownloadPath = determineJarDownloadPath();
                 final File newDetectJar = installOrUpdateScanner(jarDownloadPath);
                 if (newDetectJar != null) {
+                    List<String> arrlist = new LinkedList<>(Arrays.asList(args));
+                    arrlist.add("--selfUpdated");
+                    args = arrlist.toArray(args);
```

Comment:
If self update already updated Detect once, we want to inform subsequent Detect invocation about it so that it need not check for self update again.

Reply:
This might be obvious but I like this as a comment in the code itself.

Reply:
Done. Thanks